CHANGES_UNRELEASED.md

```diff
@@ -18,3 +18,10 @@ Use the template below to make assigning a version number during the release cut
   - Description of the change with a link to the pull request ([#0000](https://github.com/mozilla/application-services/pull/0000))
 
 -->
+## Push
+### What's changed
+  - The push component will not attempt to auto-recover from the server losing its UAID ([#4347](https://github.com/mozilla/application-services/pull/4347))
+    - The push component will return a new kotlin Error `UAIDNotRecognizedError` in cases where auto-recovering isn't possible (when subscribing)
```

Comment:
Some context on why we don't try to auto-recover on `subscribe` calls, but do so on `verifyConnection`, (I'm open to being convinced with other suggestions!).

If the consumer (A-C) is calling `verifyConnection` they're expecting to get a list of the channels they should re-subscribe to if something went wrong

but if they are calling `subscribe`, that expectation isn't there. So if we attempt to wipe the UAID and the channels, the consumer would not be notified to resubscribe.

The way I'm thinking of making this better is to let the consumer (A-C) catch this error and run `verifyConnection` fixing its worldview.